# Complete functional mapping of infection- and vaccine-elicited antibodies against the fusion peptide of HIV
### Adam S. Dingens, Priyamvada Acharya, Hugh K. Haddox, Reda Rawi, Kai Xu, Gwo-Yu Chuang, Hui Wei, John R. Mascola,  Bridget Carragher, Clint S. Potter, Julie Overbaugh, Peter D. Kwong*, Jesse D. Bloom*

Experiments analyzed here were performed by Adam Dingens in the [Bloom lab](http://research.fhcrc.org/bloom/en.html) and [Overbaugh lab](https://research.fhcrc.org/overbaugh/en.html) in the summer/autumn of 2017. 


Here, we use mutational antigenic profiling to comprehensively map escape from `N123-VRC34.01`, `2712-vFP16.02`, and `2716-vFP20.01` (subsequently named `VRC34`, `FP16-02`, and `FP20-01` in this iPython notebook) using **BG505.T332N** mutant Env virus libraries.The generation and characterization of the BG505 mutant virus libraires is described in detail in [Haddox, Dingens et al. eLife 2018](https://elifesciences.org/articles/34420).

[Dingens et al. Cell Host & Microbe 2017](http://dx.doi.org/10.1016/j.chom.2017.05.003) details mutational antigenic profiling of PGT151 using BF520 Env mutant virus libraries. 

We use [dms_tools2](https://jbloomlab.github.io/dms_tools2/) to analyze the data.  This notebook does the complete analysis of the mutational antigenic profiling data. It dowloads reads from the [Sequence Read Archive](http://www.ncbi.nlm.nih.gov/sra), processes this  data, and then analyzes the selection in the context of each antibody. 

The barcoded subamplicon Illumina sequencing approach is described in detail [here](https://jbloomlab.github.io/dms_tools2/bcsubamp.html), and the computation of `differential selection` is described [here](https://jbloomlab.github.io/dms_tools2/diffsel.html).


## Organization
The mutational antigenic profiling analysis is performed by the iPython notebook [`analysis_notebook.ipynb`](analysis_notebook.ipynb). 

Subdirectories:

   * `./results/` iss generated by [`analysis_notebook.ipynb`](analysis_notebook.ipynb). It contains the `codoncounts` and `differential selection` data and raw output for all figures.
   
   * `./results/FASTQ_files/` contains the input Illumina deep sequencing data. This file is generated by [`analysis_notebook.ipynb`](analysis_notebook.ipynb), which downloads the sequencing files from the [Sequence Read Archive](http://www.ncbi.nlm.nih.gov/sra).

   * `./data/` contains all input data needed to run [`analysis_notebook.ipynb`](analysis_notebook.ipynb)). Files are described in the iPython notebok when used. 


## Results and Conclusions
All of the results are breifly detailed in the [`analysis_notebook.ipynb`](analysis_notebook.ipynb) notebook; click on this notebook to see the results.



